hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/RatisPipelineUtils.java

```diff
@@ -146,19 +165,37 @@ private static void callRatisRpc(List<DatanodeDetails> datanodes,
         SecurityConfig(ozoneConf));
     final TimeDuration requestTimeout =
         RatisHelper.getClientRequestTimeout(ozoneConf);
-    datanodes.parallelStream().forEach(d -> {
-      final RaftPeer p = RatisHelper.toRaftPeer(d);
-      try (RaftClient client = RatisHelper
-          .newRaftClient(SupportedRpcType.valueOfIgnoreCase(rpcType), p,
-              retryPolicy, maxOutstandingRequests, tlsConfig, requestTimeout)) {
-        rpc.accept(client, p);
-      } catch (IOException ioe) {
-        String errMsg =
-            "Failed invoke Ratis rpc " + rpc + " for " + d.getUuid();
-        LOG.error(errMsg, ioe);
-        exceptions.add(new IOException(errMsg, ioe));
-      }
-    });
+    try {
+      POOL.submit(() -> {
+        datanodes.parallelStream().forEach(d -> {
+          final RaftPeer p = RatisHelper.toRaftPeer(d);
+          try (RaftClient client = RatisHelper
+              .newRaftClient(SupportedRpcType.valueOfIgnoreCase(rpcType), p,
+                  retryPolicy, maxOutstandingRequests, tlsConfig,
+                  requestTimeout)) {
+            rpc.accept(client, p);
+          } catch (IOException ioe) {
+            String errMsg =
+                "Failed invoke Ratis rpc " + rpc + " for " + d.getUuid();
+            LOG.error(errMsg, ioe);
+            exceptions.add(new IOException(errMsg, ioe));
+          }
+        });
+      }).get();
+    } catch (ExecutionException ex) {
+      LOG.error("Execution exception occurred during createPipeline", ex);
+      throw new IOException("Execution exception occurred during " +
+          "createPipeline", ex);
+    } catch (RejectedExecutionException ex) {
+      LOG.error("RejectedExecutionException, occurred during " +
```

Comment:
Same here, don't log.

Reply:
Logging here, because in the actual createPipelines method in BackGroudPipelineCreator which calls this method when we return IOException they break from while loop. That is the reason for the logging. Let me know if we still don't want to log here?

Reply:
That method is already logging right: 
```
          pipelineManager.createPipeline(type, factor);
        } catch (IOException ioe) {
          break;
        } catch (Throwable t) {
          LOG.error("Error while creating pipelines {}", t);
          break;
        }
```
So if we log here the exception will be logged twice. Perhaps I'm looking at it wrong.

Reply:
We return IOException, so in case of IOException we have not logging right?
It has only break;